lib/dolly/property.rb

```diff
@@ -21,7 +21,7 @@ def cast_value(value)
     end
 
     def custom_class(value)
-      value = value.is_a?(Hash) ? value.symbolize_keys : value
+      value = value.is_a?(Hash) ? hash_with_indifferent_access_value(value) : value
```

Comment:
do we want to change a explicit hash value to a HashWithI...?? this sound shady

Reply:
right, its explicit it should keep that value and not be changed

Reply:
In your sample with `contracts` in `School` is `contracts defined in properties as `Hash` or `HashWithIndifferentAccess` ?

Reply:
but this means changing all models to use HashWith... in the class_name.

Reply:
its defined as `property :contracts, class_name: Hash, default: Hash.new`

Reply:
now that i think about it, dolly is already changing hashes to HashWithIndifferent, this was just missing that transformation.

Reply:
you are right, but then this is weird... as there will be cases where it makes more sense a Hash, and we are force-casting that property to HWIA,   right now the default is HWIF, for all data that comes in, but properties are casted to the class that we set on the property method, that way people con define the right type for the property they want. Force casting Hash to HWIA, even if the document is parsed as such is confusing...

Reply:
But, then again, a HWIA is a Hash, and dolly is setting now everything as string keys by default, so ok... I guess we can use this.  Just want to tested on BS and some other projects that sue newer dolly and see if there arent any side effects